job-runner/runner/runner.go

```diff
@@ -190,6 +188,34 @@ TRY_LOOP:
 	}
 }
 
+// Actually run the job.
+func (r *runner) runJob(jobData map[string]interface{}) (startTime, finishTime int64, ret job.Return, err error) {
+	var startedAt, finishedAt int64
+
+	defer func() {
+		// Recover from a panic inside Job.Run()
+		if panicErr := recover(); panicErr != nil {
+			// Set return values.
+			startTime = startedAt
+			finishTime = time.Now().UnixNano()
+			ret = job.Return{
+				State: proto.STATE_FAIL,
+				Exit:  1,
+			}
+			err = fmt.Errorf("panic from job.Run: %s", panicErr)
```

Comment:
This makes it into JLE.Error? If so, probably good to make a quick code comment here to let future devs know why this line is here (it seems isolated and nothing doing anything, but it's important).